### PR TITLE
Fixed scancode with modifiers matching in InputEventKey

### PR DESCRIPTION
**Pull request by Chaosus**
I have a problem when implementing  multiple keys for one action. If you set the first key with any modifier(for example CTRL), and than second key without CTRL, the second key will be ignored because it will be identical to first. This will resolve it

Update: so let me rephrase it -

I want to apply two keys to one action - one with modifier the other without. I do it using `InputMap.action_add_event` method. This method using the match comparsion, and currently not allow adding input keys with identical scancode even if modifiers are different, I think its bug and should be fixed. Without this change, I need to create separate action which I do not want cuz its overkill.

### CODE REVIEW

**Comment by Chaosus** [comment]
OOps maybe is_pressed should be pressed... sorry for invalid PR :/

**Comment by Chaosus** [comment]
@akien-mga what do you think about it ?

**Comment by akien-mga** [comment]
As discussed with @reduz, this is sensitive code and a more thorough solution is needed to properly support key combinations in actions (and handle this use case too). This will be worked on for 3.2, so closing this PR for now.
